compendium/tools/SCRCodeGen/test/ReferenceAutogenFiles.hpp

```diff
@@ -57,11 +57,14 @@ extern "C" US_ABI_EXPORT void DeleteInstance_DSSpellCheck_SpellCheckImpl(scd::Co
 #include <vector>
 #include <cppmicroservices/ServiceInterface.h>
 #include "cppmicroservices/servicecomponent/detail/ComponentInstanceImpl.hpp"
+#include "cppmicroservices/BundleInitialization.h"
 #include "SpellCheckerImpl.hpp"
 
 namespace sc = cppmicroservices::service::component;
 namespace scd = cppmicroservices::service::component::detail;
 
+CPPMICROSERVICES_INITIALIZE_BUNDLE
```

Comment:
on further thought, having SCRCodeGen put in the `CPPMICROSERVICS_INITIALIZE_BUNDLE` macro automatically is unnecessary. Bundle authors already have to add `CPPMICROSERVICS_INITIALIZE_BUNDLE` to the bundle's source code and its reasonable to expect bundle authors to continue to do so for DS bundles. 

Reverting the SCRCodeGen change also simplifies the CMake build scripts; now there is no distinction needed on whether a DS bundle is being built and there are no concerns about duplicate symbols coming from multiple uses of `CPPMICROSERVICS_INITIALIZE_BUNDLE`.

Reply:
While it is, strictly speaking, unnecessary to add it to the code generator, it is also unnecessary to make every bundle author add it manually. If the purpose of DS is to eliminate boilerplate, and this is boilerplate, then I think it should be done by DS. It feels like a technical debt that bundle authors still need to write this, and this project seems like a perfect time to address it.

While the volume of changed lines is large, that is due to repetition - the change itself is actually quite small. If you are concerned about the number of changed lines, I can invert the flag, so the default is DS and only a few bundles need to specify that they *don't* use DS. But that feels a bit counterintuitive and doesn't seem to solve any problem.

Reply:
I agree that `CPPMICROSERVICS_INITIALIZE_BUNDLE` is boilerplate and I like your thinking of eliminating technical debt as you make changes, however the cost of keeping this as boilerplate is low. DS mainly eliminates the need for a user to implement a `BundleActivator` class. This class is the biggest potential source of bugs generated by users and eliminating its use through DS has significant value. On the other hand,  `CPPMICROSERVICS_INITIALIZE_BUNDLE` is a single line of code where the potential bug comes from the user not adding it to their source code. Users who are using CppMicroServices are already adding `CPPMICROSERVICS_INITIALIZE_BUNDLE` and the goal of this change is not to eliminate more boilerplate code.

I am less concerned about the # of changes to the CMakeLists.txt files and more concerned about the fact that developers have to remember to add another parameter to `usFunctionCreateTestBundleWithResources` when creating DS test bundles. If the logic was inverted, the same concern remains; adding another parameter to remember. I would prefer to see another CMake function created specifically for creating DS test bundles instead of adding a parameter onto an existing one.

I see less upside to auto-generating `CPPMICROSERVICS_INITIALIZE_BUNDLE` than not auto-generating it. Consider an existing user of CppMicroServices who already has build infrastructure to build their CppMicroServices modules. Its likely they will have to make changes to their infrastructure to make sure there isn't a re-definition coming from `CPPMICROSERVICS_INITIALIZE_BUNDLE` with this change. Whereas if this doesn't exist, no infrastructure change is necessary.  I am valuing not forcing potential build infrastructure changes upon users over removing a piece of boilerplate code.